docs/topics/workflow.md

```diff
@@ -0,0 +1,74 @@
+# Workflow in ICEkit
```

Comment:
Overall seems good. Would love a test implementation on a default model (Page or Article) as part of this review.